### PR TITLE
listener print strings

### PR DESCRIPTION
**Pull request by dagar**
Example

```Console
nsh> listener log_message

TOPIC: log_message instance 0 #1
log_message_s
        timestamp: 4128786 (55.251776 seconds ago)
        severity: 3
        text: "[pmw3901] driver start failed"

nsh> listener task_stack_info

TOPIC: task_stack_info instance 0 #1
task_stack_info_s
        timestamp: 128244081 (0.845543 seconds ago)
        stack_free: 428
        task_name: "px4io"
```

### CODE REVIEW

File: src/modules/load_mon/load_mon.cpp [hunk 1]
```diff
@@ -247,12 +247,13 @@ void LoadMon::_stack_usage()
...
-		if (system_load.tasks[task_index].valid && system_load.tasks[task_index].tcb->pid > 0) {
+		if (system_load.tasks[task_index].valid && (system_load.tasks[task_index].tcb->pid > 0)) {
 
 			stack_free = up_check_tcbstack_remain(system_load.tasks[task_index].tcb);
 
-			strncpy((char *)task_stack_info.task_name, system_load.tasks[task_index].tcb->name,
-				task_stack_info_s::MAX_REPORT_TASK_NAME_LEN);
+			static_assert(sizeof(task_stack_info.task_name) == CONFIG_TASK_NAME_SIZE,
+				      "task_stack_info.task_name must match NuttX CONFIG_TASK_NAME_SIZE");
+			strncpy((char *)task_stack_info.task_name, system_load.tasks[task_index].tcb->name, CONFIG_TASK_NAME_SIZE);
```
Comment (julianoes) [inline src/modules/load_mon/load_mon.cpp]: I think newer compilers are unhappy if you copy `sizeof` the char array and you need to copy `sizeof - 1` and make sure the end byte is `\0`.

Comment (dagar) [reply src/modules/load_mon/load_mon.cpp]: Updated.

**Comment by dagar** [comment]
@julianoes can you try this again with your particular toolchain?

**Comment by julianoes** [comment]
Builds fine with GCC 9.1.1 and clang 8.0.0.

I'll check at home on Arch as well later but I doubt it's newer.

**Pull request merged**